Name: Device Menu Megamix with Media Volume Control
Version: 1.4-1
Author: Jason Robitaille
Description: Removes the date and battery level, adds brightness & media volume control, adds phone submenu with radio/data/roaming toggles, adds GPS toggle, removes the airplane mode toggle, and adds a flashlight led toggle, on the device menu  (MIT license)

--- .orig/usr/lib/luna/system/luna-systemui/app/controllers/devicemenu-assistant.js
+++ /usr/lib/luna/system/luna-systemui/app/controllers/devicemenu-assistant.js
@@ -33,6 +33,7 @@
 		this.controller.get('palm-device-menu').addClassName('palm-touch-rows-'+Mojo.Environment.DeviceInfo.touchableRows);
 			
 		this.controller.setupWidget('brightSlider', {minValue: 1, maxValue: 100}, this.brightModel = { value: 50});
+		this.controller.setupWidget('mvolSlider', {minValue: 0, maxValue: 100}, this.mvolModel = { value: 50});
 						
 		this.modesModel = {scrollbars: false, mode: "vertical"},
 		this.controller.setupWidget('devicescroller', {}, this.modesModel);
@@ -40,6 +41,7 @@
 		
 		this.drawerModel = {myOpenProperty:false};
 		this.controller.setupWidget('brightdetails', {modelProperty:'myOpenProperty'}, this.drawerModel);
+		this.controller.setupWidget('mvoldetails', {modelProperty:'myOpenProperty'}, this.drawerModel);
 		this.controller.setupWidget('phonedetails', {modelProperty:'myOpenProperty'}, this.drawerModel);
 		this.controller.setupWidget('wifidetails', {modelProperty:'myOpenProperty'}, this.drawerModel);
 		this.controller.setupWidget('btdetails', {modelProperty:'myOpenProperty'}, this.drawerModel);
@@ -118,6 +120,8 @@
 	activate: function() {
 		this.controller.get('brightSlider').addEventListener(Mojo.Event.propertyChange, this.handleBrightSlider.bindAsEventListener(this));
 		this.controller.get('brightRow').addEventListener(Mojo.Event.tap, this.handleBrightTap.bindAsEventListener(this));
+		this.controller.get('mvolSlider').addEventListener(Mojo.Event.propertyChange, this.handleMvolSlider.bindAsEventListener(this));
+		this.controller.get('mvolRow').addEventListener(Mojo.Event.tap, this.handleMvolTap.bindAsEventListener(this));
 		this.controller.get('phoneRow').addEventListener(Mojo.Event.tap, this.handlePhoneTap.bindAsEventListener(this));
 		this.controller.get('phone_radio').addEventListener(Mojo.Event.tap, this.togglePhone.bindAsEventListener(this));
 		this.controller.get('phone_data').addEventListener(Mojo.Event.tap, this.toggleData.bindAsEventListener(this));
@@ -153,6 +157,18 @@
 		this.controller.get('dm_brightness').innerText = "Brightness: " + Math.round(event.value) + "%";
 	},
 	
+	handleMvolTap: function(event) {
+		this.controller.get('mvoldetails').mojo.setOpenState(!this.controller.get('mvoldetails').mojo.getOpenState());
+	},
+	
+	handleMvolSlider: function(event) {
+		this.controller.serviceRequest('palm://com.palm.audio/media', {
+			method: 'setVolume',
+			parameters:{volume:Math.round(event.value)}
+		});
+		this.controller.get('dm_mvol').innerText = "Media Volume = " + Math.round(event.value);
+	},
+	
 	handlePhoneTap: function(event) {
 		this.controller.get('phonedetails').mojo.setOpenState(!this.controller.get('phonedetails').mojo.getOpenState());
 	},
@@ -1161,6 +1177,7 @@
 			this.wifidrawer.mojo.setOpenState(false);
 		}		
 		this.controller.get('brightdetails').mojo.setOpenState(false);
+		this.controller.get('mvoldetails').mojo.setOpenState(false);
 		this.controller.get('phonedetails').mojo.setOpenState(false);
 		//this.isVisible = false;
 		if(this.isVisible){
@@ -1218,6 +1235,19 @@
 					}
 				}.bind(this)
 			});
+			this.controller.serviceRequest('palm://com.palm.audio/media', {
+				method: 'getVolume',
+				parameters:{},
+				onSuccess: function(response) {
+					if(response.volume != undefined) {
+						this.mvolModel.value = parseInt(response.volume);
+						this.controller.modelChanged(this.mvolModel);
+						this.controller.get('dm_mvol').innerText = "Media Volume = " + response.volume;
+					} else {
+						this.controller.get('dm_mvol').innerText = "Media Volume = unknown";
+					}
+				}.bind(this)
+			});
 			this.controller.serviceRequest('palm://com.palm.telephony', {
 				method: 'powerQuery',
 				parameters: {},
--- .orig/usr/lib/luna/system/luna-systemui/app/views/devicemenu/devicemenu-scene.html
+++ /usr/lib/luna/system/luna-systemui/app/views/devicemenu/devicemenu-scene.html
@@ -29,6 +29,20 @@
 				<div class="palm-section-divider">
 				</div> 
 
+				<div class="palm-row first" id="mvolRow" x-mojo-tap-highlight='momentary'><div class="palm-row-wrapper">
+					<div id="dm_mvol" class="title">&nbsp;
+					</div>
+				</div></div>
+
+				<div id="mvoldetails" x-mojo-element="Drawer">
+					<div class="palm-row first"><div class="palm-row-wrapper">
+						<div x-mojo-element="Slider" id="mvolSlider" style="padding-top: 30px; height: 50px;"></div>
+					</div></div>
+				</div>
+
+				<div class="palm-section-divider">
+				</div> 
+
 				<div class="palm-row first" id="phoneRow" x-mojo-tap-highlight='momentary'><div class="palm-row-wrapper">
 					<div id="dm_phone" class="title">Phone</div>
 				</div></div>
